benches/cub/device/merge_sort/custom_less_thrust_comparison.cu

```diff
@@ -0,0 +1,85 @@
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+#include <thrust/random.h>
+#include <thrust/shuffle.h>
+
+#include <thrust/detail/temporary_array.h>
+
+#include <cub/device/device_merge_sort.cuh>
+
+/*
+ * Arithmetic types with the custom comparator lead to merge sort usage.
+ */
+
+class less_comparator
+{
+public:
+  template <typename T>
+  __device__ bool operator()(T i, T j) noexcept
+  {
+    return i < j;
+  }
+};
+
+template <typename T,typename Derived>
+thrust::detail::temporary_array<T, Derived> create_tmp_storage(
+  thrust::execution_policy<Derived> &policy, std::size_t size)
+{
+  return thrust::detail::temporary_array<T, Derived>(policy, size);
+}
+
+template <typename T>
+void custom_less(nvbench::state &state, nvbench::type_list<T>)
+{
+  const auto elements = static_cast<long long int>(state.get_int64("Elements"));
+
+  thrust::device_vector<T> data(elements);
+  thrust::sequence(data.begin(), data.end());
+
+  thrust::default_random_engine rng;
+
+  state.add_element_count(elements);
+  state.add_global_memory_reads<T>(elements);
+  state.add_global_memory_writes<T>(elements);
+
+  size_t temp_size = 0;
+  cub::DeviceMergeSort::SortKeys(
+    nullptr,
+    temp_size,
+    thrust::raw_pointer_cast(data.data()),
+    elements,
+    less_comparator());
+
+  using namespace nvbench::exec_tag;
+  state.exec(timer | sync, // This benchmark needs a timer and syncs internally
```

Comment:
I don't think you'll need `sync` here. `thrust::shuffle` syncs, but that's outside of the timer's critical section.